src/scored.rs

```diff
@@ -33,22 +33,15 @@ impl<K: PartialOrd, T> Ord for MinScored<K, T> {
     fn cmp(&self, other: &MinScored<K, T>) -> Ordering {
         let a = &self.0;
         let b = &other.0;
-        if a == b {
-            Ordering::Equal
-        } else if a < b {
-            Ordering::Greater
-        } else if a > b {
-            Ordering::Less
-        } else {
-            // these are the NaN cases
-            if a != a && b != b {
-                Ordering::Equal
-            } else if a != a {
-            // Order NaN less, so that it is last in the MinScore order
-                Ordering::Less
-            } else {
-                Ordering::Greater
-            }
+        match b.partial_cmp(a) {
```

Comment:
It doesn't matter much, but I don't think this one is wanted -- unless  I misremember, calling partial_cmp on floats is pretty terrible performance wise (and the explicit operators better).